src/v2/view-builder/views/idp/AuthenticatorIdPView.js

```diff
@@ -0,0 +1,20 @@
+import EnrollIdPBody from './EnrollAuthenticatorIdPBody';
+import ChallengeIdpBody from './ChallengeAuthenticatorIdPBody';
+import BaseAuthenticatorView from '../../components/BaseAuthenticatorView';
+import AuthenticatorEnrollFooter from '../../components/AuthenticatorEnrollFooter';
+
+export default BaseAuthenticatorView.extend({
+  initialize () {
+    BaseAuthenticatorView.prototype.initialize.apply(this, arguments);
+
+    const currentAuthenticator = this.options.appState.get('currentAuthenticator');
+    const authenticatorEnrollments = this.options.appState.get('authenticatorEnrollments')?.value || [];
+    authenticatorEnrollments.filter(authenticator => authenticator.id === currentAuthenticator.id);
+
+    // Set the view based on whether or not we're in a challenge flow
+    this.Body = authenticatorEnrollments.length
```

Comment:
aha... this is interesting.
For other authenticators, they are `challenge-authenticator` and `authenticator-enrollment-data/enroll-authenticator` respectively for challenge and enroll.
I'm curious why we don't use same pattern (I meant same remediation forms) then either backend decide to redirect to IDP (responds 302) or still widget handle the redirect (`when form name is challenge-authenticator and authenticator is external idp, redirect`)

Reply:
Glad this caught your attention!

The IdP Authenticator is intended [not to use the `idp/idx/challenge/answer` endpoint](https://oktawiki.atlassian.net/wiki/spaces/~ajay.thomas/pages/1749414459/IDP+Authenticator+on+OIE+Assurance?focusedCommentId=1776779962#Enrollment-flow) as we do not control the verification logic. I believe 

When discussing this with the backend team, it sounds like they wanted to avoid the additional forms completely - treating "redirects" as a redirect form.

> when form name is challenge-authenticator and authenticator is external idp, redirect

This is interesting, but wouldn't it require us to check multiple objects inside of the `remediation` response before acting?